response.go

```diff
@@ -111,17 +111,33 @@ func (r *Response) apiError() error {
 		b.statusCode = r.StatusCode
 		return &b
 	}
-	if r.StatusCode > 299 {
-		return fmt.Errorf("%s (%d)", http.StatusText(r.StatusCode), r.StatusCode)
```

Comment:
This is where I think we should keep the `> 299` and instead of returning the `fmt.Errorf`, return your `httpError(r.StatusCode)`.